buildSrc/src/main/groovy/com/google/firebase/gradle/plugins/FirebaseLibraryExtension.java

```diff
@@ -14,22 +14,67 @@
 
 package com.google.firebase.gradle.plugins;
 
+import com.google.common.collect.ImmutableSet;
 import com.google.firebase.gradle.plugins.ci.device.FirebaseTestLabExtension;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
 import javax.inject.Inject;
 import org.gradle.api.Action;
-import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.GradleException;
+import org.gradle.api.Project;
+import org.gradle.api.UnknownDomainObjectException;
 
 public class FirebaseLibraryExtension {
+  private final Project project;
+  private final Set<FirebaseLibraryExtension> librariesToCoRelease = new HashSet<>();
+
+  /** Indicates whether the library has public javadoc. */
   public boolean publishJavadoc = true;
+
+  /** Indicates whether sources are published alongside the library. */
   public boolean publishSources;
+
+  /** Firebase Test Lab configuration/ */
   public final FirebaseTestLabExtension testLab;
 
   @Inject
-  public FirebaseLibraryExtension(ObjectFactory objectFactory) {
-    this.testLab = new FirebaseTestLabExtension(objectFactory);
+  public FirebaseLibraryExtension(Project project) {
```

Comment:
I would go out of the way to document that this is not equivalent to -HEAD dependencies that folks might be used to. The key distinction why it may not be extended to fit that use case might be that the libraries being co-released are at the exact same version.

Reply:
I updated the javadoc of the `releaseWith` method.